tests/test_bootpolicy.robot

```diff
@@ -0,0 +1,187 @@
+*** Settings ***
+
+Documentation   This testsuite is for testing boot policy function.
+
+Resource        ../lib/rest_client.robot
+Resource        ../lib/ipmi_client.robot
+Resource        ../lib/utils.robot
+
+Suite Setup     Open Connection And Log In
+Suite Teardown  Close All Connections
+
+
+*** Variables ***
+${SYSTEM_SHUTDOWN_TIME}    ${5}
+
+
+*** Test Cases ***
+
+Set Onetime boot order using REST
+    [Documentation]   This testcase is to set onetime boot policy using REST
+    ...               URI and then verify using REST API and ipmitool.\n
+   
+    ${bootpolicy} =   Set Variable   ONETIME
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   ONETIME
+    ${output} =    Run IPMI Standard command   chassis bootparam get 5
+    Should Contain   ${output}   Options apply to only next boot
+
+Set Permanent boot order using REST
+    [Documentation]   This testcase is to set permanent boot policy using REST
+    ...               URI and then verify using REST API and ipmitool.\n
+   
+    ${bootpolicy} =   Set Variable   PERMANENT
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   PERMANENT
+    ${output} =    Run IPMI Standard command   chassis bootparam get 5
+    Should Contain   ${output}   Options apply to all future boots
+
+Set Onetime boot order using IPMITOOL
+    [Documentation]   This testcase is to set boot policy to onetime boot using ipmitool
+    ...               and then verify using REST URI and ipmitool.\n
+
+    Run IPMI command   0x0 0x8 0x05 0x80 0x00 0x00 0x00 0x00
+    Read the Attribute   /org/openbmc/settings/host0   boot_policy
+    Response Should Be Equal   ONETIME
+    ${output} =   Run IPMI Standard command   chassis bootparam get 5
+    Should Contain   ${output}   Options apply to only next boot
+
+   
+Set Permanent boot order using IPMITOOL
+    [Documentation]   This testcase is to set boot policy to permanent using ipmitool
+    ...               and then verify using REST URI and ipmitool.
+
+    Run IPMI command   0x0 0x8 0x05 0xC0 0x00 0x00 0x00 0x00
+    Read the Attribute   /org/openbmc/settings/host0   boot_policy
+    Response Should Be Equal   PERMANENT
+    ${output} =   Run IPMI Standard command   chassis bootparam get 5
+    Should Contain   ${output}   Options apply to all future boots
+
+Boot order with permanent boot policy
```

Comment:
No it will not change. We need to boot  to OS to see it change. Booting to OS is not possible to do at present in automation.  I added this test case to make sure to nothing is changing with permanent setting.